src/git-dot/ocaml_deflate.ml

```diff
@@ -1 +1 @@
-../git-unix/ocaml_deflate.ml
```

Comment:
I don't undertand this commit. Why did you put everything under `utils`? Some of the files need to be linked with `lwt.unix`, others with `compiler-libs`, etc.

Reply:
I put all implementation in one directory and use `ln -sfn ...` in `git-unix`, `git-dot`, etc. to use these implementations to avoid any redundant code. For example, `fs_lwt_unix.ml` need to be available in the `git-dot` folder, `git-unix` and `git-top`, so we share a common implementation of the file-system available in one directory.

It's same for `sha1.ml` (need to be available in any sub-project).

The `utils` directory is not a project as `git-unix`, it does not contain any `jbuild` file and does not produce any libraries or binaries, it's just a bulk of implementations needed to make Git's functors.

IMO, it's better to do this for maintainability and avoid to apply a fix/change a multiple times for a same implementation. If we found a bug in `fs_lwt_unix.ml`, we fix it and this fix is directly available for other sub-project which use this specific implementation.

Reply:
Yikes. Please move the core/common files into the "core" library (e.g. `git`) as it used to be and use it directly -- you should never need to either symlink or copy/duplicate files.

The `fs_lwt_unix.ml` is used by the unix backend only move it into the `git-unix` library. etc.